pkg/fake_pub_server/lib/fake_pub_server.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:fake_gcloud/mem_datastore.dart';
+import 'package:fake_gcloud/mem_storage.dart';
+import 'package:gcloud/db.dart';
+import 'package:logging/logging.dart';
+import 'package:pub_server/repository.dart';
+import 'package:pub_server/shelf_pubserver.dart';
+import 'package:shelf/shelf.dart' as shelf;
+
+import 'package:pub_dartlang_org/account/backend.dart';
+import 'package:pub_dartlang_org/frontend/backend.dart';
+import 'package:pub_dartlang_org/frontend/email_sender.dart';
+import 'package:pub_dartlang_org/frontend/handlers.dart';
+import 'package:pub_dartlang_org/frontend/static_files.dart';
+import 'package:pub_dartlang_org/frontend/upload_signer_service.dart';
+import 'package:pub_dartlang_org/history/backend.dart';
+import 'package:pub_dartlang_org/scorecard/backend.dart';
+import 'package:pub_dartlang_org/scorecard/scorecard_memcache.dart';
+import 'package:pub_dartlang_org/shared/analyzer_client.dart';
+import 'package:pub_dartlang_org/shared/configuration.dart';
+import 'package:pub_dartlang_org/shared/handler_helpers.dart';
+import 'package:pub_dartlang_org/shared/package_memcache.dart';
+import 'package:pub_dartlang_org/shared/redis_cache.dart';
+import 'package:pub_dartlang_org/shared/storage.dart';
+
+final _logger = Logger('fake_server');
+
+class FakePubServer {
```

Comment:
Okay, this is a good start... IMO it would great if we had a dependency injection to the point that we didn't need to do this in a separate package... But let's not open that can of worms now.